test/integration_test.go

```diff
@@ -233,6 +233,23 @@ func TestDetailedExitCodeChangesPresentAll(t *testing.T) {
 	assert.Equal(t, 2, exitCode.Get())
 }
 
+func TestDetailedExitCodeFailOnFirstRun(t *testing.T) {
+	t.Parallel()
+
+	testFixturePath := filepath.Join(testFixtureDetailedExitCode, "fail-on-first-run")
+
+	helpers.CleanupTerraformFolder(t, testFixturePath)
+	tmpEnvPath := helpers.CopyEnvironment(t, testFixturePath)
+
+	var exitCode tf.DetailedExitCode
+	ctx := context.Background()
+	ctx = tf.ContextWithDetailedExitCode(ctx, &exitCode)
+
+	_, _, err := helpers.RunTerragruntCommandWithOutputWithContext(t, ctx, "terragrunt run-all plan --terragrunt-log-level trace --terragrunt-non-interactive -detailed-exitcode --terragrunt-working-dir "+util.JoinPath(tmpEnvPath, testFixturePath))
```

Comment:
Don't fix this now, but we should remember to replace this with a `run --all` version for future compatibility.

- [ ] TODO: Replace `run-all` with `run --all` in `TestDetailedExitCodeFailOnFirstRun`